### PR TITLE
feat: improve CLI error messages

### PR DESCRIPTION
**Pull request by ammario**
Before clibase:
Screen Shot 2023-03-24 at 12 37 48 PM

clibase:
Screen Shot 2023-03-24 at 12 38 22 PM

now:
image

### CODE REVIEW

**Review by mafredri — COMMENTED** [review]
I agree that the original error for login is pretty bad, but I don't think reformatting it improves it, it's just a bad message to begin with (which is a side-effect of how it's implemented). It should simply have said `Unable to reach "https://dev.coder.con", is the URL correct and is Coder accessible from your browser?` (i.e. mainly omit the first user part and the detailed error part).

We could have a `--debug` flag that prints more detailed error messages.

Errors from the API still seem a bit messy:

```
❯ /tmp/coder ssh list
ERROR
GET http://127.0.0.1:3000/api/v2/workspaces?q=owner%3Ame: unexpected status code 401: You are signed out or your session has expired. Please sign in again to continue.: Try logging in using 'coder login <url>'.
    Error: API key expired at "2023-03-21 15:14:13.352559 +0000 UTC".
```

They were better before:

```
❯ coder list
You are signed out or your session has expired. Please sign in again to continue. Try logging in using 'coder login <url>'.
Run 'coder list --help' for usage.
```

Unless debugging, the expiry date isn't very useful, and the status code is pretty much useless since the human message already says it. I think keeping the error message mostly to a single line is good for CLI users too.

Personally I think the formatting doesn't really help here either:

```
❯ /tmp/coder ssh test -G
ERROR
upload GPG public keys and ownertrust to workspace
─────
export local ownertrust from GPG
─────
`gpg --export-ownertrust` failed: stderr: gpg: Fatal: can't open '/home/mafredri/.gnupg/trustdb.gpg': No such file or directory

stdout:

exit status 2
─────
exit status 2
─────
```

But the original wasn't good either:

```
❯ coder ssh test -G
upload GPG public keys and ownertrust to workspace: export local ownertrust from GPG: `gpg --export-ownertrust` failed: stderr: gpg: Fatal: can't open '/home/mafredri/.gnupg/trustdb.gpg': No such file or directory

stdout:

exit status 2: exit status 2
Run 'coder ssh --help' for usage.
```

Original feels less busy though, but obviously stdout should be discarded since it's empty (and likely always), as should the double exit status.

**Comment by ammario** [comment]
kk.. I got rid of decoration:

image

and simplified in cases where we have a `codersdk.Error`:
Screen Shot 2023-03-24 at 3 38 58 PM

**Comment by ammario** [comment]
I want to avoid changing the errors themselves in this PR and just get us to a state where we feel they're as good as they were before clibase.

**Comment by ammario** [comment]
Screen Shot 2023-03-24 at 4 16 02 PM

Per slack poll, it's now on one line

**Comment by mafredri** [comment]
Looks better 👍🏻, thanks.

I do seem to have a knack for finding these problems though. Might be good to verify what the behavior is for a few more commands:

```
❯ /tmp/coder tokens ls
You are signed out or your session has expired. Please sign in again to continue.Try logging in using 'coder login <url>'.: You are signed out or your session has expired. Please sign in again to continue.Try logging in using 'coder login <url>'.
```

I.e. duplicated message and `continue.Try` without space.

And weird long-indent for second exit code 🤔
```
❯ /tmp/coder ssh test -G
upload GPG public keys and ownertrust to workspace: export local ownertrust from GPG: `gpg --export-ownertrust` failed: stderr: gpg: Fatal: can't open '/home/mafredri/.gnupg/trustdb.gpg': No such file or directory

stdout:

exit status 2:                                                                                                                 exit status 2
```

**Comment by ammario** [comment]
Good catch. We'll need golden files for errors soon...

Fixed both by removing the unnecessary recursion in the error formatter.. c9e31bb5d

File: cli/root.go [hunk 1]
```diff
@@ -822,89 +818,61 @@ func isConnectionError(err error) bool {
...
-			),
-		)
+	var msg string
+	var sdkError *codersdk.Error
+	if errors.As(err, &sdkError) {
+		// We don't want to repeat the same error message twice, so we
+		// only show the SDK error on the top of the stack.
+		msg = sdkError.Message
+		if sdkError.Helper != "" {
+			msg = msg + "\n" + sdkError.Helper
```
Comment (mafredri) [inline cli/root.go:848]: The newline here makes sense and makes this look nice. 👍🏻

**Review by mafredri — APPROVED** [review]
LGTM!

**Pull request merged**